### PR TITLE
Add kube config generator for clusters

### PR DESCRIPTION
**Pull request by shydefoo**
This PR adds common code to generate k8s rest.Config that will be used in Merlin and Turing.

### CODE REVIEW

File: api/pkg/cluster/cluster.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+// GenerateConfig generates a k8s rest.Config that is used to instantiate k8s clients
+// ie any client that is generated by client-go generators
+func (k *K8sCredsManager) GenerateConfig() (*rest.Config, error) {
+	restConf := generateKubeConfig(k.K8sConfig)
+	r, err := sigyaml.Marshal(restConf)
+	if err != nil {
+		return nil, err
+	}
+	if err != nil {
```
Comment (mbruner) [inline api/pkg/cluster/cluster.go:80]: Duplication

File: api/pkg/cluster/cluster_test.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+		input     K8sConfig
+		output    rest.Config
+	}{
+		{
+			name:      "Test basic server, ca cert",
+			wantError: false,
+			input: K8sConfig{
+				Name: "dummy-cluster",
+				Cluster: &clientcmdapiv1.Cluster{
+					Server:                "https://123.456.789",
```
Comment (mbruner) [inline api/pkg/cluster/cluster_test.go:24]: unrealistic IP

**Comment by shydefoo** [comment]
Note: Interface name and methods have been renamed to improve readability

File: api/pkg/cluster/cluster.go (hunk 1 above)
Comment (shydefoo) [reply api/pkg/cluster/cluster.go]: Removed

File: api/pkg/cluster/cluster_test.go [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+	tests := []struct {
+		name      string
+		wantError bool
+		input     K8sConfig
+		output    *clientcmdapiv1.Config
+	}{
+		{
+			name:      "Test basic server, ca cert",
+			wantError: false,
+			input: K8sConfig{
```
Comment (krithika369) [inline api/pkg/cluster/cluster_test.go:67]: Do we really need this test case, or does testing the overarching public method `.ToRestConfig()` adequately test this functionality too? Going by https://martinfowler.com/articles/practical-test-pyramid.html#WhatToTest

It seems both these test cases are working on the same input, just that this test is validating the intermediate result which I'm not sure is of much value.

**Review by krithika369 — APPROVED** [review]
Left one small comment, the rest LGTM. Thanks, @shydefoo !

File: api/pkg/cluster/cluster_test.go (hunk 3 above)
Comment (shydefoo) [reply api/pkg/cluster/cluster_test.go:67]: TIL, yep that makes sense, will remove it.

File: api/pkg/cluster/cluster_test.go (hunk 2 above)
Comment (shydefoo) [reply api/pkg/cluster/cluster_test.go]: changed to use example domain name

**Pull request merged**